packages/nvidia_nat_memmachine/tests/test_memmachine_api_calls.py

```
@@ -0,0 +1,619 @@
+        self.calls = []
+        self.return_values = {}
+    
+    def record_call(self, method_name: str, args: tuple, kwargs: dict):
+        """Record an API call."""
+        self.calls.append({
+            'method': method_name,
+            'args': args,
+            'kwargs': kwargs
+        })
+    
+    def get_calls(self, method_name: str = None):
```

Ruff (0.14.13)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)